File: modules/infra/src/main/java/hm/orz/chaos114/android/slideviewer/infra/model/Slide.java
package hm.orz.chaos114.android.slideviewer.infra.model;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import java.io.Serializable;

import lombok.Data;

@Data
@DatabaseTable
public class Slide implements Serializable {
    @DatabaseField(generatedId = true)
    private int id;

    @DatabaseField(foreign = true, foreignAutoRefresh = true, columnName = "talk_id")
    private Talk talk;

    @DatabaseField
    private String original;

    @DatabaseField
    private String preview;

    @DatabaseField
    private String thumb;
}

File: modules/infra/src/main/java/hm/orz/chaos114/android/slideviewer/infra/model/Slide.kt
package hm.orz.chaos114.android.slideviewer.infra.model

import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable

@DatabaseTable
data class Slide(
        @DatabaseField(generatedId = true)
        var id: Int = 0,

        @DatabaseField(foreign = true, foreignAutoRefresh = true, columnName = "talk_id")
        var talk: Talk? = null,

        @DatabaseField
        var original: String? = null,

        @DatabaseField
        var preview: String? = null,

        @DatabaseField
        var thumb: String? = null
)
